sympy/geometry/point.py

```diff
@@ -261,59 +296,127 @@ def affine_rank(*points):
             raise TypeError("Must have at least one point")
 
         # make sure we're genuinely points
-        # and translate to every point to the origin
-        origin = Point.pointify(points[0])
-        points = [Point.pointify(p, dimension=origin.ambient_dimension) - origin for p in points[1:]]
+        # and translate every point to the origin
+        points = Point.normalize_dimensions(None, *[
+            Point.pointify(i) for i in points])
+        origin = points[0]
+        points = [p - origin for p in points[1:]]
 
         m = Matrix([p.args for p in points])
         return m.rank()
 
     @staticmethod
     def project(a, b):
         """Project the point `a` onto the non-zero point `b`."""
-        a, b = Point.pointify(a), Point.pointify(b)
+        a, b = Point.normalize_dimensions(None,
+            Point.pointify(a), Point.pointify(b))
         if b.is_zero:
             raise ValueError("Cannot project to the zero vector.")
-        if a.ambient_dimension != b.ambient_dimension:
-            raise ValueError("Points must be of the same dimension")
 
         return b*(a.dot(b) / b.dot(b))
 
     @staticmethod
-    def pointify(a, dimension=None, ensure_geometry_entity=True, ensure_point=True, **kwargs):
-        """If `a` is a `Point` or a non-iterable, it is left alone.
-        Otherwise, it is converted to a point with `kwargs` passed along
-        to `Point()`. If `dimension` is set (default `None`), a dimension check is carried out.
-        If `ensure_geometry_entity` is set (default `True`), a type check for GeometryEntity
-        is carried out and the return value will be a GeometryEntity.
-        If `ensure_point` is set (default `True`), a type check for Point is carried out
-        and the return value will be a Point."""
+    def pointify(a, dimension=None, strict=False, ensure_point=True):
```

Comment:
The Point could be padded or trimmed. In the most recent pass through I have eliminated `pointify` and moved that logic into `Point`. `Point*D` are just stubs to call `Point` to do the heavy lifting.